### PR TITLE
Bump up python versions

### PR DESCRIPTION
**Pull request by Alnusjaponica**
## Motivation
Resolve https://github.com/optuna/optuna/issues/3964 and https://github.com/optuna/optuna-examples/issues/164

## Description of the changes
- Add CI job for python 3.11, (and 3.12 if possible).
- Remove version specification of torch and its families in `pytorch/requirements.txt` and `rl/requirements.txt`, due to the impossibility of installing the same version of the necessary libraries in both Python 3.7 and 3.11.
See https://github.com/pytorch/pytorch/releases and https://stable-baselines3.readthedocs.io/en/master/misc/changelog.html for reference. 
~Please note that the CI job for Python 3.7 has been removed in `pytorch` and `rl.~

### CODE REVIEW

**Comment by contramundum53** [comment]
@not522 Could you review this PR?

File: pytorch/requirements.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ plotly
 pytorch-ignite
 pytorch-lightning
 skorch
-torch
-torchvision
-torchaudio
+torch==2.0.0
+torchvision==0.15.1
+torchaudio==2.0.1
```
Comment (not522) [inline pytorch/requirements.txt:8]: How about leaving these without version specification? I think that it should work.

Comment (Alnusjaponica) [reply pytorch/requirements.txt]: Sounds reasonable and it might enable the tests run in python 3.7.

**Review by not522 — APPROVED** [review]
LGTM!

**Pull request merged**